### PR TITLE
Document double nat scenario for OpenStack deployments

### LINKED ISSUE #805: Document steps and diagram showcasing Double NAT traversal scenario for deployments in openstack
When 2 clusters are deployed on OSP environment, they do not have direct reach-ability over public IP. Hence adding a public-ip and annotating the gateway nodes is necessary. please help adding doc for covering this deployment scenario. 

slack discussion details: https://coreos.slack.com/archives/C0134E73VH6/p1657802632620539?thread_ts=1657703762.382189&cid=C0134E73VH6

### PR DESCRIPTION
**Pull request by sridhargaddam**
Fixes: https://github.com/submariner-io/submariner-website/issues/805
Signed-off-by: Sridhar Gaddam <sgaddam@redhat.com>

### CODE REVIEW

File: src/content/operations/nat-traversal/_index.en.md [hunk 1]
```diff
@@ -356,9 +356,29 @@ NAT, or if other applications on the internal network were already using the 450
 ports, the remote ends would not be able to contact gateway C or D over the expected ports.
 {{% /notice %}}
 
-### Double NAT Traversal
+### Double NAT Traversal (scenario 1)
 
-In this case case, A & B cluster gateways have direct reachability over their private
+In this case, clusters C & D are neither reachable on their private IPs (192.168.0.4 and 192.168.0.4) nor
+on the public-ip. However, they are reachable over the private floating IPs (10.2.0.1 and 10.2.0.2).
```
Comment (nyechiel) [inline src/content/operations/nat-traversal/_index.en.md]:
```suggestion
on the public IP. However, they are reachable over the private floating IPs (10.2.0.1 and 10.2.0.2).
```

File: src/content/operations/nat-traversal/_index.en.md [hunk 2]
```diff
@@ -356,9 +356,29 @@ NAT, or if other applications on the internal network were already using the 450
...
+Submariner cannot detect these private floating IPs. To get the connectivity working, you can annotate
+the Gateway node with the private-floating IP as shown below.
```
Comment (nyechiel) [inline src/content/operations/nat-traversal/_index.en.md]:
```suggestion
the Gateway node with the private floating IP as shown below.
```

File: src/content/operations/nat-traversal/_index.en.md [hunk 3]
```diff
@@ -356,9 +356,29 @@ NAT, or if other applications on the internal network were already using the 450
...
+kubectl annotate node $GWD gateway.submariner.io/public-ip=ipv4:10.2.0.2
+
+# restart the gateways to pick up the new setting
+for cluster in C D;
+do
+  kubectl delete pod -n submariner-operator -l app=submariner-gateway --kubeconfig $cluster
+done
+```
+
+### Double NAT Traversal (scenario 2)
```
Comment (nyechiel) [inline src/content/operations/nat-traversal/_index.en.md:379]: The diagram is missing the cluster names (C and D).

**Review by nyechiel — CHANGES_REQUESTED** [review]

File: src/content/operations/nat-traversal/_index.en.md (hunk 3 above)
Comment (sridhargaddam) [reply src/content/operations/nat-traversal/_index.en.md:379]: Good point, I missed it. Thanks.

**Review by nyechiel — APPROVED** [review]

**Review by aswinsuryan — APPROVED** [review]

**Review by dfarrell07 — APPROVED** [review]

File: src/content/operations/nat-traversal/_index.en.md [hunk 4]
```diff
@@ -356,9 +356,29 @@ NAT, or if other applications on the internal network were already using the 450
 ports, the remote ends would not be able to contact gateway C or D over the expected ports.
 {{% /notice %}}
 
-### Double NAT Traversal
+### Double NAT Traversal (scenario 1)
 
-In this case case, A & B cluster gateways have direct reachability over their private
+In this case, clusters C & D are neither reachable on their private IPs (192.168.0.4 and 192.168.0.4) nor
```
Comment (yboaron) [inline src/content/operations/nat-traversal/_index.en.md:361]: Shouldn't it be 192.168.128.4 ?

File: src/content/operations/nat-traversal/_index.en.md [hunk 5]
```diff
@@ -356,9 +356,29 @@ NAT, or if other applications on the internal network were already using the 450
...
+do
+  kubectl delete pod -n submariner-operator -l app=submariner-gateway --kubeconfig $cluster
+done
+```
+
+### Double NAT Traversal (scenario 2)
+
+In this case, A & B cluster gateways have direct reachability over their private
 IPs (10.0.0.1 and 10.1.0.1) possibly with large MTU capabilities, while between cluster
 C and D (192.168.0.4 and 192.168.0.4 too), reachability over the private IPs is not possible
```
Comment (yboaron) [inline src/content/operations/nat-traversal/_index.en.md:383]: Same here, it should be '(192.168.0.4 and 192.168.128.4 too)'

File: src/content/operations/nat-traversal/_index.en.md (hunk 4 above)
Comment (nyechiel) [reply src/content/operations/nat-traversal/_index.en.md:361]: Why? The diagram uses 192.168.0.4

Comment (sridhargaddam) [reply src/content/operations/nat-traversal/_index.en.md:361]: Private IPs are the node-ips. As per the diagram its 192.168.0.4 (deliberately shown the same IPs as its possible in such deployments) and the private floating-ips are 10.2.0.1 and 10.2.0.2 respectively. The public-ip auto-discovery for such clusters would be the same for both (2.1.1.1).

File: src/content/operations/nat-traversal/_index.en.md [hunk 6]
```diff
@@ -356,9 +356,29 @@ NAT, or if other applications on the internal network were already using the 450
...
 
-In this case case, A & B cluster gateways have direct reachability over their private
+In this case, clusters C & D are neither reachable on their private IPs (192.168.0.4 and 192.168.0.4) nor
+on the public IP. However, they are reachable over the private floating IPs (10.2.0.1 and 10.2.0.2).
+Submariner cannot detect these private floating IPs. To get the connectivity working, you can annotate
+the Gateway node with the private floating IP as shown below.
+
+![on-premises](/images/natt/double_nat_scenario1.png)
+
+```bash
```
Comment (yboaron) [inline src/content/operations/nat-traversal/_index.en.md:368]: Q : the annotated public IP (e.g: 10.2.0.1) will be used as EP's public IP for all connections (including to clusters C,D) ,  can you elaborate how the connection from cluster A to C will work ?

Comment (sridhargaddam) [reply src/content/operations/nat-traversal/_index.en.md:368]: There is no Cluster A in this topology. Please look at the next scenario and it is currently not supported in Submariner.

File: src/content/operations/nat-traversal/_index.en.md (hunk 4 above)
Comment (yboaron) [reply src/content/operations/nat-traversal/_index.en.md:361]: Ooops, I missed the diagram

File: src/content/operations/nat-traversal/_index.en.md (hunk 6 above)
Comment (yboaron) [reply src/content/operations/nat-traversal/_index.en.md:368]: Ack, 10x

**Review by yboaron — APPROVED** [review]

**Pull request merged**